ovs/extensions/healthcheck/generic_hc.py

```diff
@@ -439,3 +441,21 @@ def verify_rabbitmq(result_handler):
                 result_handler.failure('RabbitMQ has partition issues: {0}'.format(', '.join(partitions)))
         else:
             result_handler.skip('RabbitMQ is not running/active on this server!')
+
+    @staticmethod
+    @expose_to_cli(MODULE, 'recovery-domain-test', HealthCheckCLIRunner.ADDON_TYPE)
+    def check_recovery_domains(result_handler):
+        prim_domains = [i.name for i in DomainList.get_domains() if len(i.storage_router_layout['regular']) >= 1]
+        for domain in DomainList.get_domains():
+            layout = domain._storage_router_layout()
+            recovery = layout['recovery']
+            regular = layout['regular']
+            # Check recovery usage
+            if len(recovery) >= 1 and domain.name not in prim_domains:
+                result_handler.warning('warning: domain {0} set as recovery, but not as regular domain'.format(domain.name))
```

Comment:
No need to prepend the 'warning' to the message. The result handler does this for you.
Also casing!

Reply:
Another thing. List all the storagerouters for whom this domain is set this way. This information is normally stored within the recovery variable(so: for storagerouter_guid in recovery: StorageRouter(storagerouter_guid) ... )